probe-rs/src/rtt/channel.rs

```diff
@@ -16,14 +16,172 @@ pub trait RttChannel {
     fn buffer_size(&self) -> usize;
 }
 
+#[repr(C)]
+#[derive(Debug, FromZeroes, FromBytes, Copy, Clone)]
+pub(crate) struct RttChannelBufferInner<T> {
+    header: RttChannelBufferInnerHeader<T>,
+    buffer_offset: RttChannelBufferInnerBufferOffset<T>,
+    flags: T,
+}
+
+impl<T> RttChannelBufferInner<T> {
+    pub fn write_buffer_ptr_offset(&self) -> usize {
+        std::mem::size_of::<RttChannelBufferInnerHeader<T>>()
+    }
+
+    pub fn read_buffer_ptr_offset(&self) -> usize {
+        std::mem::size_of::<RttChannelBufferInnerHeader<T>>() + std::mem::size_of::<T>()
```

Comment:
This is pretty hard to understand because the second `size_of` points in the middle of an inner `struct`. Although `offset_of_nested` is not stable yet in Rust, this might still be a similarly good place to use `offset_of!`, like `std::mem::offset_of!(Self, buffer_offset) + std::mem::offset_of!(RttChannelBufferInnerBufferOffset<T>, read_offset)`

Reply:
Thank you for sharing your good method!

I rewrote `*_offset` functions using `offset_of!`.
047338571e5de163c8ecc48bef5e24b90dad7c22